source/slang/slang-compiler.cpp

```diff
@@ -964,6 +964,23 @@ namespace Slang
         typedef DownstreamCompiler::CompileOptions CompileOptions;
         CompileOptions options;
 
+        /* Let's set the compiler specific options 
+
+          We can only do this if the endToEndReq is set. */
```

Comment:
This might be worth fixing in the long run, since downstream compiler arguments seem like the kind of thing that could arguably belong on a `TargetReq`, and thus could be used by any compile including API-based ones.